util/entropy.c

```diff
@@ -17,56 +18,150 @@
  * XXX win32 where there is no concept of /dev/urandom.
  */
 
+struct entropy_read
```

Comment:
The `{` should be on the same line as this.

Also, it kind of feels like this should be `entropy_reader`, in the sense that a structure is a noun rather than a verb.  Up to you, just thought I'd mention it.

Reply:
While I'm here, I feel like this structure deserves some sort of comment.  Something along the lines of "Entropy reader state.  At present it holds a file descriptor open to /dev/urandom but in the future this structure may gain other OS-dependent state, e.g. a Windows Handle.".

Reply:
Added a REBASE commit.  I went with `struct entropy_read_cookie`.  (I was sorely tempted to call it `entropy_cookie`, but since the functions all start with `entropy_read`, I couldn't justify it to myself)

I tweaked your proposed comment with "file descriptor for /dev/urandom", since at the moment that we allocate the cookie, the file descriptor is not (yet) open to /dev/urandom.